Makefile

```diff
@@ -37,6 +37,12 @@ $(SELPROJ): $(TOOLS_BIN_DIR) $(TOOLS_DIR)/go.mod
 	cd $(TOOLS_DIR) && $(GO) build -tags tools -o bin/selproj github.com/aiven/go-utils/selproj
 
 
+MOCKERY := $(TOOLS_BIN_DIR)/mockery
+
+$(MOCKERY): $(TOOLS_BIN_DIR) $(TOOLS_DIR)/go.mod
+	cd $(TOOLS_DIR) && $(GO) build -o bin/mockery github.com/vektra/mockery/v2
+
```

Comment:
Mockery has a huge list of dependencies we don't use. That might increase the eventual file size of the provider. Moved it to tools.